app/components/at_mention/at_mention.test.js

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Clipboard, Platform} from 'react-native';
+
+import mattermostManaged from 'app/mattermost_managed';
+
+import Preferences from 'mattermost-redux/constants/preferences';
+
+import AtMention from './at_mention';
+
+jest.mock('react-intl');
+
+describe('AtMention', () => {
+    const baseProps = {
+        isSearchResult: false,
+        mentionName: 'user-1',
+        mentionStyle: {color: '#2389d7'},
+        navigator: {push: jest.fn(), showModal: jest.fn()},
+        onLongPress: jest.fn(),
+        onPostPress: jest.fn(),
+        textStyle: {color: '#3d3c40', fontSize: 15, lineHeight: 20},
+        teammateNameDisplay: 'username',
+        theme: Preferences.THEMES.default,
+        usersByUsername: {'user-1': {id: 'user-1', username: 'username_1'}},
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(
+            <AtMention {...baseProps}/>
+        );
+
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should match snapshot, for user without username', () => {
+        const usersByUsername = {'user-1': {id: 'user-1'}};
+        const wrapper = shallow(
+            <AtMention
+                {...baseProps}
+                usersByUsername={usersByUsername}
+            />
+        );
+
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should match snapshot, with suffix', () => {
+        const usersByUsername = {'user-1': {id: 'user-1', username: 'user'}};
+        const wrapper = shallow(
+            <AtMention
+                {...baseProps}
+                usersByUsername={usersByUsername}
+            />
+        );
+
+        expect(wrapper.getElement()).toMatchSnapshot();
+    });
+
+    test('should call Clipboard.setString on handleCopyMention', () => {
+        Clipboard.setString = jest.fn();
+        const wrapper = shallow(
+            <AtMention {...baseProps}/>
+        );
+
+        const username = 'user-1';
+        wrapper.setState({username});
+        wrapper.instance().handleCopyMention();
+        expect(Clipboard.setString).toHaveBeenCalledTimes(1);
+        expect(Clipboard.setString).toBeCalledWith(`@${username}`);
+    });
+
+    test('should call props.onLongPress and mattermostManaged.getLocalConfig on handleLongPress', async () => {
```

Comment:
Name not descriptive. I can't tell what this is trying to test. Therefore if it breaks I will just delete it.

Reply:
So lets see if I understand this: when the LongPress on the username it calls the getLocalConfig from MattermostManaged to determine if the copy protection was set by an EMM in order to add or not the option to Copy the mention in the tooltip menu and then calls the `onLongPress` from the props to actually show the tooltip menu.

I'm I right?